client/src/actions/actions.js

```diff
@@ -0,0 +1,91 @@
+import {
+    ADD_SONG,
+    BEGIN_DOWNLOAD_SONG_INFO,
+    DELETE_SONG, DOWNLOAD_SONG_INFO_ERROR, ERROR_CONFIRMED,
+    UPDATE_LINK,
+    UPDATE_SONG
+} from "./action-types";
+import config from "../config";
+import {formatString, getVideoIdFromLink} from "../services/formatting";
+
+export const updateLink = (newLink) => {
+    return {
+        type: UPDATE_LINK,
+        link: newLink
+    };
+};
+
+export const addSong = (link) => (dispatch) => {
```

Comment:
This is not really an action.
I would say this is a resolver case, but it does a `fetch` request.
You should put this near the `thunk`  code of this project

Reply:
what do you mean the `thunk` code of the project?

Reply:
I thought you have thunk code to address the server, but you only use the `thunk` midllewere.
You can remove `thunk` from this project all together.
What I meant is that you can't have big action like addressing the server in the `redux` code - run the fetch out of the redux code and add the result a variable to this method

Reply:
I need thunk to perform async actions.

Uncaught Error: Actions must be plain objects. Use custom middleware for async actions.

Reply:
I got it

Reply:
seems ok